### PR TITLE
Add a class to check and update choice types to the modifications section

### PR DESCRIPTION
**Pull request by okeefm**
This PR adds a class called `ChoiceTypeCreator` to the data script. This class checks the bundles generated by the data script for conformance to Must Support choice types, as specified in US Core. If a particular choice type is missing from the bundles for a given resource/profile pair, the `ChoiceTypeCreator` will either: 
a) choose resource(s) to swap to the new choice type, if enough exist that we can swap resources and not get rid of other choices in that choice list for that resource/profile pair
or 
b) clone resources to include the additional needed choice types, if there aren't enough resources of that resource/profile pair to be able to swap them.

### CODE REVIEW

File: lib/choice_type_creator.rb
Comment (Jammjammjamm) [inline lib/choice_type_creator.rb:96]: This part of the code needs more real variable names.

File: lib/choice_type_creator.rb
Comment (Jammjammjamm) [inline lib/choice_type_creator.rb:66]: Rather than `get_resources_by_class`, why not just iterate over the profiles in `MUST_SUPPORT_CHOICE_TYPES` and have a `select` to grab all of the resources that fit a particular profile?

File: lib/choice_type_creator.rb
Comment (Jammjammjamm) [inline lib/choice_type_creator.rb:73]: I think it would be nice if everything in here was done by an instance rather than the class. So `ChoiceTypeCreator.check_choice_types` iterates over the profiles and instantiates an instance of `ChoiceTypeCreator` to do the real work. An instance provides a lot more options for cleaning up the code (e.g. breaking up this big method by replacing comments explaining what code does with methods whose names say what the code does).

File: lib/modifications.rb
Comment (Jammjammjamm) [inline lib/modifications.rb:563]: 😱

File: lib/modifications.rb
Comment (Jammjammjamm) [inline lib/modifications.rb:215]: Don't forget this.

**Review by Jammjammjamm — COMMENTED** [review]
Feel free to ignore my big refactor comments if you don't think it's worth it.

```
Traceback (most recent call last):
	4: from uscore-data-script.rb:151:in `<main>'
	3: from uscore-data-script/lib/modifications.rb:539:in `modify!'
	2: from uscore-data-script/lib/choice_type_creator.rb:66:in `check_choice_types'
	1: from uscore-data-script/lib/choice_type_creator.rb:138:in `get_resources_by_class'
uscore-data-script/lib/choice_type_creator.rb:138:in `merge': wrong number of arguments (given 5, expected 1) (ArgumentError)
```

**Comment by okeefm** [comment]
@Jammjammjamm the error is because I was using a function that's only available in Ruby 2.7. I'll see if I can figure out a workaround that should work in 2.5.6

**Comment by Jammjammjamm** [comment]
Yeah, I'm not getting that error with 2.7.2. Just add `.ruby-version` and `.tool-versions` files?

**Comment by okeefm** [comment]
Works for me, as long as people in the project are willing to install/use a different Ruby version from what Inferno uses.

**Comment by Jammjammjamm** [comment]
We could also bring inferno up. I investigated migrating it to 2.7.2, and it required minimal effort.

**Comment by okeefm** [comment]
Yeah, that might be worth doing. Ruby 2.5 goes EoL in March 2021.

File: lib/choice_type_creator.rb
Comment (okeefm) [reply lib/choice_type_creator.rb:66]: Yeah, that would be another option. I think it might be a little harder to index into the `MUST_SUPPORT_CHOICE_TYPES` hash if the resources were collected by profile URL rather than class (though I guess you could use the `.class` of the object to get the key of the hash.

Comment (Jammjammjamm) [reply lib/choice_type_creator.rb:66]: Is it necessary to index into the hash? I was imagining something like:
```
MUST_SUPPORT_CHOICE_TYPES.each do |_, must_supports|
  must_supports[:profiles].each do |profile|
    all_the_bundled_resources
      .select { |resource| resource.meta.profile == profile }
      .each { |resource| process these resources }
  end
end
```

Comment (okeefm) [reply lib/choice_type_creator.rb:66]: I think this would be feasible. I'll investigate during the refactor for the comment below.

**Review by arscan — APPROVED** [review]
Generated and ran against inferno, looks good.

**Comment by arscan** [comment]
I see you rently rebased -- are you good on this now?

**Comment by okeefm** [comment]
I think this is good to go, yes (pending @Jammjammjamm 's approval). We decided to hold off on the refactor, in the interest of getting the code in.

**Review by Jammjammjamm — APPROVED** [review]

**Pull request merged**
